piccolo3/client/PiccoloCoolboxControl.py

```diff
@@ -116,6 +116,96 @@ def current_voltage(self):
         return self._current_voltage
 
 
+class PiccoloCurrent(PiccoloNamedClientComponent):
+    """manage current control on coolbox"""
+
+    NAME = "coolboxctrl"
+
+    def __init__(self, baseurl, name):
+
+        super().__init__(baseurl, name)
+
+        self._current_current = None
+
+        self._callbacks = []
+
+        self.add_task(self._get_current_current())
+
+    def register_callback(self, cb):
+        self._callbacks.append(cb)
+
+    async def _get_current_current(self):
+        u = 'current_current'
+        async for s in self.a_observe(u):
+            for cb in self._callbacks:
+                await cb(json.dumps((self.name, u, s)))
+            self._current_current = s
+
+    async def get_current_current(self):
+        self.log.info(self.uri('current_current'))
+        self._current_current = await self.a_get('current_current')
+        return self._current_current
+
+    @property
+    def current_current(self):
+        return self._current_current
+
+
+class PiccoloFan(PiccoloNamedClientComponent):
+    """manage fan control on coolbox"""
+
+    NAME = "coolboxctrl"
+
+    def __init__(self, baseurl, name):
+
+        super().__init__(baseurl, name)
+
+        self._target_fan_state = None
+        self._current_fan_state = None
+
+        self._callbacks = []
+
+        self.add_task(self._get_target_fan_state())
```

Comment:
just noticed this - does the fan have a target state? shouldn't that be just  the state which can be on or off?

Reply:
You're right, I think I've over engineered this. Obviously the fan can either be on or off, but there is a difference between whether it should be on, and it actually being on, which is what I was doing here.  I agree though, this is unnecessary, I'll refactor.